src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -76,37 +91,191 @@ static DamageTable D_809901F8 = {
+    return false;
+}
+
+void EnTg_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTg* this = THIS;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (limbIndex == HONEY_AND_DARLING_LIMB_MAN_HEAD) {
+        Matrix_MultVec3f(&zeroVec, &this->actor.focus.pos);
+    }
+}
+
+void EnTg_Draw(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+    GraphicsContext* gfxCtx;
+
+    Matrix_Push();
+    EnTg_DrawHeart(play, this->effects, ARRAY_COUNT(this->effects));
+    Matrix_Pop();
+
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Gfx_EnvColor(play->state.gfxCtx, 0, 50, 160, 0));
+    gSPSegment(POLY_OPA_DISP++, 0x09, Gfx_EnvColor(play->state.gfxCtx, 255, 255, 255, 0));
+
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          EnTg_OverrideLimbDraw, EnTg_PostLimbDraw, &this->actor);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+/**
+ * Spawns a heart at the first effects array index that's not enabled.
+ * Because of the frame counts, only two hearts are ever spawned at a time.
+ */
+void EnTg_SpawnHeart(EnTg* this, EnTgHeartEffect* effect, Vec3f* heartStartPos, s32 numEffects) {
+    Vec3f heartVelocity = { 0.0f, 1.5f, 0.0f };
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+
+    for (i = 0; i < numEffects && effect->isEnabled; i++, effect++) {}
+
+    if (i < numEffects) {
+        effect->isEnabled = true;
+        effect->pos = *heartStartPos;
+        effect->velocity = heartVelocity;
+        effect->unusedZeroVec = zeroVec;
+        effect->scale = 0.01f;
+        effect->pos.x += 4.0f * Math_SinS(this->actor.shape.rot.y);
+        effect->pos.z += 4.0f * Math_CosS(this->actor.shape.rot.y);
+        effect->timer = 16;
+    }
+}
+
+/**
+ * The heart path is curvy as it floats up because of the use of Math_SinS and Math_CosS.
+ */
+void EnTg_UpdateHearts(PlayState* play, EnTgHeartEffect* effect, s32 numEffects) {
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s16 yaw = Camera_GetInputDirYaw(GET_ACTIVE_CAM(play));
+    s32 i;
+
+    for (i = 0; i < numEffects; i++, effect++) {
+        if (effect->isEnabled == true) {
+            if (DECR(effect->timer) == 0) {
+                effect->isEnabled = false;
+            }
+            effect->pos.y += effect->velocity.y;
+            effect->pos.x += 2.0f * Math_SinS(effect->angle);
+            effect->pos.z += 2.0f * Math_CosS(effect->angle);
+
+            Matrix_Push();
+            Matrix_Translate(effect->pos.x, effect->pos.y, effect->pos.z, MTXMODE_NEW);
+            Matrix_RotateYS(yaw, MTXMODE_APPLY);
+            Matrix_MultVec3f(&zeroVec, &effect->pos);
+            Matrix_Pop();
+
+            effect->angle += 0x1770;
+        }
+    }
+}
+
+void EnTg_DrawHeart(PlayState* play, EnTgHeartEffect* effect, s32 numEffects) {
```

Comment:
```suggestion
void EnTg_DrawHearts(PlayState* play, EnTgHeartEffect* effect, s32 numEffects) {
```
Small nit, but this will draw all enabled hearts, not just a single one.